src/config/ConfigBuilder.ts

```diff
@@ -91,15 +95,85 @@ export class ConfigBuilder {
             } else if (key === 'customLogger') {
                 this.handleLogger(value);
             } else if (key === 'customCredentials') {
-                this.effectiveConfig.customCredentials = jsonObject;
```

Comment:
🤔 Why was it like that? Was it a bug?

Reply:
Yes, that was mistakenly changed to this in https://github.com/hazelcast/hazelcast-nodejs-client/pull/1080. Since we had no tests for the custom credentials, we failed to catch it there